Re-enable: Shortids for Datastores and DDSes (#24231)
microsoft/FluidFramework: .changeset/deep-mangos-stand.md

---
"@fluidframework/container-runtime": minor
"@fluidframework/datastore": minor

> "@fluid-private/test-end-to-end-tests": minor

Reviewer: This package should not be part of the changet because it's private.

Author: Fixed

> ---
> "@fluidframework/container-runtime": minor
> "@fluidframework/datastore": minor
> "@fluid-private/test-end-to-end-tests": minor
> ---
> ---
> "section": feature
> ---

Reviewer 2: ```suggestion
---
"@fluidframework/container-runtime": minor
"@fluidframework/datastore": minor
"__section": feature
---
```

The changeset format recently changed, but since this was created before the biold-tools update, we can just fix it manually.

Author: Done

> Shorter ids for Datastores and DDSes

Reviewer 2: ```suggestion
Shorter IDs for Datastores and DDSes
```

Is Datastore the name of the code entity or is it DataStore, etc?

Author: Fixed. It's DataStore.

> Fixed a long pending bug, which now enables us to use shorter ids for Datastores and DDSes when `enableRuntimeIdCompressor:"on"` is set in `IContainerRuntimeOptions`

Reviewer 2: ```suggestion
The Fluid Framework will now use shorter IDs for Datastores and DDSes when `enableRuntimeIdCompressor:"on"` is set in `IContainerRuntimeOptions`.
```

What's the benefit for customers? Space savings? Perf? Something else?

Author: Added more details about expected perf improvement.

---
"@fluidframework/container-runtime": minor
"@fluidframework/datastore": minor
"__section": feature
---

Shorter IDs for DataStores and DDSes

> Fixed a long pending bug, which now enables us to use shorter ids for Datastores and DDSes when `enableRuntimeIdCompressor:"on"` is set in `IContainerRuntimeOptions`. This change should help in reduction in summary and snapshot sizes as well as improve runtime performance because of smaller memory footprint.

Reviewer 2: ```suggestion
Fluid Framework will now use shorter IDs for Datastores and DDSes when `enableRuntimeIdCompressor:"on"` is set in `IContainerRuntimeOptions`. This change should help reduce summary and snapshot sizes as well as improve runtime performance because of a smaller memory footprint.
```

Author: Done
